bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,559 @@
+                    registry_id=registry_id, error=error.response["Error"]["Message"]
+                )
+            )
+
+
+def _get_creds_from_token(token):
+    cred_string = base64.b64decode(token).decode("ascii")
+    username, password = str(cred_string).split(":")
+    return username, password
+
+
+def _make_user_data(username, password, registry, tag):
+    base_format = """MIME-Version: 1.0
+Content-Type: multipart/mixed; boundary=\"==MYBOUNDARY==\"
+
+--==MYBOUNDARY==
+Content-Type: text/cloud-config; charset=\"us-ascii\"
+
+runcmd:
+
+- sudo yum update -y
+- sudo amazon-linux-extras install docker -y
+- sudo service docker start
+- sudo usermod -a -G docker ec2-user
+- curl 'https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip' -o 'awscliv2.zip'
+- unzip awscliv2.zip
+- sudo ./aws/install
+- ln -s /usr/bin/aws aws
+- docker login --username {0} --password {1} {2}
+- sudo docker pull {3}
+- sudo docker run -p 5000:5000 {3}
+
+--==MYBOUNDARY==--
+""".format(
+        username, password, registry, tag
+    )
+    encoded = base64.b64encode(base_format.encode("ascii")).decode("ascii")
+    return encoded
+
+
+def _make_cloudformation_template(
+    project_dir,
+    user_data,
+    s3_bucket_name,
+    sam_template_name,
+    ami_id,
+    instance_type,
+    autoscaling_min_size,
+    autoscaling_desired_size,
+    autoscaling_max_size,
+):
+    """
+    NOTE: SSH ACCESS TO INSTANCE MAY NOT BE REQUIRED
+    """
+    if (
+        autoscaling_min_size <= 0
+        or autoscaling_max_size < autoscaling_min_size
+        or autoscaling_desired_size < autoscaling_min_size
+        or autoscaling_desired_size > autoscaling_max_size
+    ):
+        raise BentoMLException(
+            "Wrong autoscaling capacity specified. "
+            "It should be min_size <= desired_size <= max_size"
+        )
+
+    template_file_path = os.path.join(project_dir, sam_template_name)
+    yaml = YAML()
+    sam_config = {
+        "AWSTemplateFormatVersion": "2010-09-09",
+        "Transform": "AWS::Serverless-2016-10-31",
+        "Description": "BentoML load balanced template template",
+        "Parameters": {
+            "AmazonLinux2LatestAmiId": {
+                "Type": "AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>",
+                "Default": ami_id,
+            },
+        },
+    }
+    yaml.dump(sam_config, Path(template_file_path))
+
+    with open(template_file_path, "a") as f:
+        f.write(
+            """\
+Resources:
+    SecurityGroupResource:
+        Type: AWS::EC2::SecurityGroup
+        Properties:
+            GroupDescription: "security group for bentoservice"
+            SecurityGroupIngress:
+                -
+                    IpProtocol: tcp
+                    CidrIp: 0.0.0.0/0
+                    FromPort: 5000
+                    ToPort: 5000
+                -
+                    IpProtocol: tcp
+                    CidrIp: 0.0.0.0/0
+                    FromPort: 22
+                    ToPort: 22
```

Comment:
I am not able to open the index (swagger UI) page from the public URL that displayed in the instance detail page.  Any idea why?

Reply:
Sorry about that,should be fixed now.

Reply:
hmm interesting, I am still not able to access them

Reply:
weird.
I am able to access from instance_public_ip:5000.
It may take some time to download docker image,can you try again?
Or ca you log in with `ec2_instance_connect` from consonle and check cloud-init's log with `sudo cat /var/log/cloud-init-output.log`

Reply:
Also, I think when user update the deployment, cloud formation will create a new instance, right?  We want to make sure the access endpoint is the same.  Maybe put ELB(elastic load balancer) as part of this deployment template?

Reply:
Should we add it in subsequents prs?
This pr is getting huge by now.

Reply:
I was having the same concerns as you.  I think it probably more important for the user to have a constant endpoint to call when they update their deployments.

I know I would hate if I update my ML deployments, and all other services need to update the request URL to that service.

Reply:
According to this doc,https://docs.aws.amazon.com/cli/latest/reference/autoscaling/update-auto-scaling-group.html#description
`If you associate a new launch configuration or template with an Auto Scaling group, all new instances will get the updated configuration. Existing instances continue to run with the configuration that they were originally launched with.`

Reply:
I think this may complicate update a bit.

Reply:
Update issue is fixed with replace policy,bu I am not sure how to attach load balancer to get static ip.
Any suggestions @yubozhao ?

Reply:
Working on attaching load balancer to scaling group.

Reply:
That's great. Looking forward to seeing these changes.

Reply:
fixed